ui/src/app/pages/admin/admin-users.tsx

```diff
@@ -133,50 +140,176 @@ export const AdminUsers = withUserProfile()(class extends React.Component<
-                  headerStyle={{width: '7%'}}/>
-          <Column field='name' header='Name' bodyStyle={{...styles.colStyle, width: '15%'}}
-                  headerStyle={{width: '15%'}} sortable={true}/>
-          <Column field='username' header='User Name' bodyStyle={{...styles.colStyle, width: '20%'}}
-                  headerStyle={{width: '20%'}} sortable={true}/>
-          <Column field='contactEmail' header='Contact Email' sortable={true}
-                  bodyStyle={{...styles.colStyle, width: '19%'}} headerStyle={{width: '19%'}}/>
-          {enableBetaAccess && <Column field='betaAccessRequestTime' header='Beta Access Time Requested' sortable={true}
-                  bodyStyle={{...styles.colStyle, width: '15%'}} headerStyle={{width: '15%'}}/>}
-          <Column field='userLockout' header='User Lockout'
-                  bodyStyle={{...styles.colStyle, width: '10%'}} headerStyle={{width: '10%'}}/>
-          <Column field='bypass' header='Bypass'
-                  bodyStyle={{...styles.colStyle, width: '10%'}} headerStyle={{width: '10%'}}/>
-          <Column field='audit' header='Audit'
-                  bodyStyle={{...styles.colStyle, width: '10%'}} headerStyle={{width: '10%'}}/>
-        </DataTable> :
-        <div>
-          Loading user profiles...
-          <SpinnerOverlay
-            overrideStylesOverlay={{alignItems: 'flex-start', marginTop: '2rem'}}/>
-        </div>}
+      {loading &&
+        <SpinnerOverlay opacity={0.6}
+                        overrideStylesOverlay={{alignItems: 'flex-start', marginTop: '2rem'}}/>
+      }
+      {!contentLoaded && <div>Loading user profiles...</div>}
+      {contentLoaded && <div>
+        <input data-test-id='search'
+               style={{marginBottom: '.5em', width: '300px'}}
+               type='text'
+               placeholder='Search'
+               onChange={e => this.debounceUpdateFilter(e.target.value)}
+        />
+        <DataTable value={this.convertProfilesToFields(users)}
+                   frozenWidth='200px'
+                   globalFilter={filter}
+                   paginator={true}
+                   rows={50}
+                   scrollable
+                   sortField={'firstRegistrationCompletionTimestamp'}
+                   style={styles.tableStyle}>
+          <Column field='name'
+                  bodyStyle={{...styles.colStyle}}
+                  filterField={'nameText'}
+                  filterMatchMode={'contains'}
+                  frozen={true}
+                  header='Name'
+                  headerStyle={{...styles.colStyle, width: '200px'}}
+                  sortable={true}
+                  sortField={'nameText'}
+          />
+          <Column field='status'
+                  bodyStyle={{...styles.colStyle}}
+                  excludeGlobalFilter={true}
+                  header='Status'
+                  headerStyle={{...styles.colStyle, width: '80px'}}
+          />
+          <Column field='institutionName'
```

Comment:
I'm not familiar with how filtering works in a DataTable.  What happens to Columns like this with neither "exclude" nor "match" ?

Really excellent feature addition btw

Reply:
I think all columns are included by default. So we'll match on name, institution, username, and contact email. My feeling is that filtering should "just work" without having to explain too much to users... if we need to index more fields for searching, or remove columns, we can do that over time.

Agree – this is a very good reason to use well-developed community components, to save us the time of building advanced functionality like this. I found myself thinking about this as I dug deeper into PopupTrigger, which appears to be built more or less from scratch. It has no tests, and no documentation... which just makes it frustrating to work with as a core UI component.